base/pkg/types.jl

```diff
@@ -37,54 +77,21 @@ function VersionSet(versions::Vector{VersionNumber})
 end
 VersionSet(versions::VersionNumber...) = VersionSet(VersionNumber[versions...])
 
-const empty_versionset = VersionSet([v"0.0",v"0.0"])
+const empty_versionset = VersionSet(VersionInterval[])
 
-function normalize!(A::VersionSet)
-    # removes empty intervals and fuses intervals without gaps
-    # e.g.:
-    #     [0.0.0,1.0.0) ∪ [1.0.0,1.5.0) ∪ [1.6.0,1.6.0) ∪ [2.0.0,∞)
-    # becomes:
-    #     [0.0.0,1.5.0) ∪ [2.0.0,∞)
-    # (still assumes that intervals are properly sorted)
-    ivals = A.intervals
-    l = length(ivals)
-    for k = l:-1:1
-        isempty(ivals[k]) && (splice!(ivals, k); l -= 1)
-    end
-    l > 1 || return A
-    lo, up, k0 = ivals[l].lower, ivals[l].upper, l
-    fuse = false
-    for k = (l-1):-1:1
-        lo1, up1 = ivals[k].lower, ivals[k].upper
-        if up1 == lo
-            fuse = true
-            lo = lo1
-            continue
-        end
-        fuse && splice!(ivals, (k+1):k0, (VersionInterval(lo, up),))
-        fuse = false
-        lo, up = lo1, up1
-        k0 = k
-    end
-    fuse && splice!(ivals, 1:k0, (VersionInterval(lo, up),))
-    return A
-end
-
-show(io::IO, s::VersionSet) = join(io, s.intervals, " ∪ ")
+show(io::IO, s::VersionSet) = isempty(s) ? print(io, "∅") : join(io, s.intervals, " ∪ ")
```

Comment:
That seems way too verbose, considering that it's going to be shown to users in backtrace error messages like these:
```
version range [0.0.0,0.2.0) ∪ [0.3.0,∞) required by package SomePackage, whose allowed version range is [0.1.0,0.1.2) ∪ [0.1.4,∞):
```
etc.